functional-test/src/test/groovy/org/rundeck/util/common/jobs/JobUtils.groovy

```diff
@@ -358,59 +377,60 @@ class JobUtils {
         tempFile.deleteOnExit()
         tempFile.path
     }
-    static void validateJobsImportAllSuccess(Map data){
-        if(data.failed && data.failed.size()>0){
-            throw new Exception("Some jobs failed on import: "+data.failed)
+
+    /**
+     * Executes the handler if at least one job import fails
+     * @param data
+     * @param failedJobsHandler
+     */
+    private static void validateJobsImportAllSuccess(data,
+                                                     Consumer<CreateJobResponse> failedJobsHandler = { List<Object> failedJobs -> throw new Exception("Some jobs failed on import: " + failedJobs) } ) {
+        if(data?.failed?.size()>0) {
+            failedJobsHandler.accept(data?.failed)
         }
     }
 
     /**
      * Imports an XML job file from a local resource into a specified project.
      *
      * @param projectName The name of the project into which the job file is to be imported.
-     * @param xmlResourcePath The resource path
+     * @param resourcePath The resource path
      * @param client      The RdClient object used to perform the HTTP request.
      * @param dupeOption  (Optional) The duplicate option for handling existing jobs. Defaults to DUPE_OPTION_DEFAULT.
      * @param contentType (Optional) The content type of the request. Defaults to CONTENT_TYPE_DEFAULT.
      * @return A Map representation of the JSON response body if the import is successful.
      *         The method checks for a successful response and a 200 HTTP status code.
      * @throws IllegalArgumentException if the imports fails
      */
-    static def jobImportFile(
-        String projectName,
-        String xmlResourcePath,
-        RdClient client,
-        String dupeOption = DUPE_OPTION_DEFAULT,
-        String contentType = CONTENT_TYPE_DEFAULT
-    ) {
-        URL resourceUrl = getClass().getResource(xmlResourcePath)
-        def pathXmlFile = resourceUrl != null ? resourceUrl.getPath() : xmlResourcePath
-        return jobImportFile(projectName, new File(pathXmlFile), client, dupeOption, contentType)
+    static def jobImportFile = (String projectName, String resourcePath, RdClient client, String dupeOption = DUPE_OPTION_DEFAULT, String contentType = CONTENT_TYPE_DEFAULT) -> {
```

Comment:
I had reverted the closure/lambda into a basic method definition, which is more idiomatic.  Is there a reason to use a lambda?

Reply:
Yes, the behaviour of `getClass().getResource(...)` differs when the code executes in a named module (as per your refactoring) and a Closure (original version). Multiple tests started to fail due to the inability to locate files on a file system after the change to a simple function.

Per the docs:  
> If this class is in a named Module then this method will attempt to find the resource in the module. This is done by delegating to the module's class loader findResource(String,String) method, invoking it with the module name and the absolute name of the resource. Resources in named modules are subject to the rules for encapsulation specified in the Module getResourceAsStream method and so this method returns null when the resource is a non-".class" resource in a package that is not open to the caller's module.

> Otherwise, if this class is not in a named module then the rules for searching resources associated with a given class are implemented by the defining class loader of the class. This method delegates to this object's class loader. If this object was loaded by the bootstrap class loader, the method delegates to ClassLoader. getSystemResource.
Before delegation, an absolute resource name is constructed from the given resource name using this algorithm: